src/Modal.js

```diff
@@ -19,6 +19,53 @@ limitations under the License.
 
 var React = require('react');
 var ReactDOM = require('react-dom');
+import sdk from './index';
+
+/**
+ * Wrap an asynchronous loader function with a react component which shows a
+ * spinner until the real component loads.
+ */
+const AsyncWrapper = React.createClass({
+    propTypes: {
+        /** A function which takes a 'callback' argument which it will call
+         * with the real component once it loads.
+         */
+        loader: React.PropTypes.func.isRequired,
+    },
+
+    getInitialState: function() {
+        return {
+            component: null,
+        }
+    },
+
+    componentWillMount: function() {
+        this._unmounted = false;
+        this.props.loader((e) => {
```

Comment:
just that `require` (or `require.ensure`, for that matter) takes a callback, rather than returning a promise. We can't easily wrap `require` with something that returns a promise, because that breaks webpack's static analysis. So if `loader` was expected to be a Promise, we would have to add some nasty boilerplate everywhere we loaded a component asynchronously.

IOW: currently we can do:
```
createDialogAsync((cb) => {require(['foo'], cb)});
```
If instead we had to pass a promise, this would become: 
```
d = q.defer(); 
require(['foo'], (m) => {d.resolve(m)});
createDialogAsync(d.promise);
```